platforms/beepy/src/i2c_puppet.rs

```diff
@@ -109,8 +111,11 @@ impl I2cPuppetClient {
     pub async fn from_registry(kernel: &'static Kernel) -> Self {
         loop {
             match I2cPuppetClient::from_registry_no_retry(kernel).await {
-                Some(port) => return port,
-                None => {
+                Ok(port) => return port,
+                Err(registry::ConnectError::Rejected(_)) => {
+                    unreachable!("the KeyboardMuxService does not return connect errors!")
+                }
+                Err(_) => {
                     // I2C probably isn't registered yet. Try again in a bit
                     kernel.sleep(Duration::from_millis(10)).await;
```

Comment:
I wonder if we should give the registry a waitqueue for "new service registered".

Reply:
that's a great idea actually, hmm!

Reply:
would prefer to address that in a follow-up, though.